magento/magento2
# issues-25010: fix CustomerIsGuest for quotes

### Description (*)
Fixed issue Quote `customer_is_guest` set to 0 for guest.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#24736 : Quote `customer_is_guest` set to 0 for guests
1. Fixes magento/magento2#23908 : The shipping information was unable to be saved. verify the input data and try again

### Manual testing scenarios (*)
1. Test: Customer Frontend
  - Enable Persistent Shopping Cart
  - Create a customer account
  - Go to my account, and add a shipping address
  - Add any product 
  - **_!Impotent:_** wait until the customer session expires (one hour by default) but no longer than Persistence Lifetime is set to 
  - Reload the page so that somewhere in the upper-right corner You get "Not You?" link next to the welcome message
  - Click on "Not You?" enter the credentials of the customer and log in
  - Go to Mini Cart end press Precede to Checkout
  - Select the address you have just added
  - Select a shipping method and click next button
  - you get no error message and can continue 
2. Test: Admin interface (ensure that is working as before after patch)
  - register customer in frontend 
  - add shipping and billig address
  - login in Backend
  - create order for customer (select customer and store)
  - you get no error message and can continue 

### Questions or comments


### Contribution checklist (*)
